src/measure_theory/group/basic.lean

```diff
@@ -253,4 +311,186 @@ end
+    rw ← this,
+    exact is_compact.haar_lt_top _ (hK.image hfsymm)
+  end,
+  open_pos := begin
+    assume U hU h'U,
+    rw map_apply hf.measurable hU.measurable_set,
+    refine (hU.preimage hf).haar_pos _ _,
+    have : f.symm '' U = f ⁻¹' U := equiv.image_eq_preimage _ _,
+    rw ← this,
+    simp [h'U],
+  end }
+
+/-- A Haar measure on a sigma-compact space is sigma-finite. -/
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance is_haar_measure.sigma_finite
+  [t2_space G] [sigma_compact_space G] [opens_measurable_space G] : sigma_finite μ :=
+⟨⟨{ set := compact_covering G,
+  set_mem := λ n, (is_compact_compact_covering G n).measurable_set,
+  finite := λ n, is_compact.haar_lt_top μ $ is_compact_compact_covering G n,
+  spanning := Union_compact_covering G }⟩⟩
+
+open_locale topological_space
+open filter
+
+/-- If the neutral element of a group is not isolated, then a Haar measure on this group has
+no atom. -/
+@[priority 100, to_additive]
+instance is_haar_measure.has_no_atoms
+  [t1_space G] [topological_group G] [locally_compact_space G] [borel_space G]
+  [ne_bot (𝓝[{(1 : G)}ᶜ] (1 : G))] :
+  has_no_atoms μ :=
+begin
+  suffices H : μ {(1 : G)} ≤ 0, by { constructor, simp [le_bot_iff.1 H] },
+  obtain ⟨K, K_compact, K_int⟩ : ∃ (K : set G), is_compact K ∧ (1 : G) ∈ interior K,
+  { rcases exists_compact_subset is_open_univ (mem_univ (1 : G)) with ⟨K, hK⟩,
+    exact ⟨K, hK.1, hK.2.1⟩ },
+  have K_inf : set.infinite K := infinite_of_mem_nhds (1 : G) (mem_interior_iff_mem_nhds.1 K_int),
+  have μKlt : μ K ≠ ∞ := (K_compact.haar_lt_top μ).ne,
+  have I : ∀ (n : ℕ), μ {(1 : G)} ≤ μ K / n,
+  { assume n,
+    obtain ⟨t, tK, tn⟩ : ∃ (t : finset G), ↑t ⊆ K ∧ t.card = n := K_inf.exists_subset_card_eq n,
+    have A : μ t ≤ μ K := measure_mono tK,
+    have B : μ t = n * μ {(1 : G)},
+    { rw ← bUnion_of_singleton ↑t,
+      change μ (⋃ (x ∈ t), {x}) = n * μ {1},
+      rw @measure_bUnion_finset G G _ μ t (λ i, {i}),
+      { simp only [tn, finset.sum_const, nsmul_eq_mul, haar_singleton] },
+      { assume x hx y hy xy,
+        simp only [on_fun, xy.symm, mem_singleton_iff, not_false_iff, disjoint_singleton_right] },
+      { assume b hb, exact measurable_set_singleton b } },
+    rw B at A,
+    rwa [ennreal.le_div_iff_mul_le _ (or.inr μKlt), mul_comm],
+    right,
+    apply ne_of_gt (haar_pos_of_nonempty_interior μ ⟨_, K_int⟩) },
+  have J : tendsto (λ (n : ℕ),  μ K / n) at_top (𝓝 (μ K / ∞)) :=
+    ennreal.tendsto.const_div ennreal.tendsto_nat_nhds_top (or.inr μKlt),
+  simp only [ennreal.div_top] at J,
+  exact ge_of_tendsto' J I,
+end
+
+/- The above instance applies in particular to show that an additive Haar measure on a nontrivial
+finite-dimensional real vector space has no atom. -/
+example {E : Type*} [normed_group E] [normed_space ℝ E] [nontrivial E] [finite_dimensional ℝ E]
+  [measurable_space E] [borel_space E] (μ : measure E) [is_add_haar_measure μ] :
+  has_no_atoms μ := by apply_instance
```

Comment:
Do we want examples in mathlib files? I don't know, maybe there are examples in other files, I may simply have overlooked them.

It looks like this adds compile time without producing an usable result (or maybe I am wrong and it doesn't). So my question, rather than review, is: is it fine to have that example, or would it be better to put this in the docstring of the instance?
It might be worth mentioning it in the docstring of the instance either way.

Reply:
Most often we don't want examples, because it adds to build time. Here, though, I included it to make sure that nothing breaks later on: this is a complicated instance, so when doing a refactoring in other parts of the library this could break this and keep unnoticed. The role of the example is to make sure this doesn't break.